### ISSUE TITLE
Problems from #880

### ISSUE THREAD

**Issue by drwells** [IBAMR/IBAMR#907]
Thanks to @fuhuifang for pointing this out to me :)

I don't fully understand the problem but it looks like #880 introduced some problems outside the ability of the test suite to catch. In particular, when running `CIB/ex1/` in 3D I get a segmentation fault and the following output:
```
+++++++++++++++++++++++++++++++++++++++++++++++++++
At beginning of timestep # 0
Simulation time is 0
Advancing hierarchy by timestep size dt = 1e-16
==21028== Conditional jump or move depends on uninitialised value(s)
==21028==    at 0x545CD7: IBAMR::CIBMethod::spreadForce(int, IBTK::RobinPhysBdryPatchStrategy*, std::ve
ctor<SAMRAI::tbox::Pointer<SAMRAI::xfer::RefineSchedule<3> >, std::allocator<SAMRAI::tbox::Pointer<SAMR
AI::xfer::RefineSchedule<3> > > > const&, double) (CIBMethod.cpp:596)                                 
==21028==    by 0x5DD1FB: IBAMR::IBExplicitHierarchyIntegrator::integrateHierarchy(double, double, int)
 (IBExplicitHierarchyIntegrator.cpp:220)                                                              
==21028==    by 0x955B96: IBTK::HierarchyIntegrator::advanceHierarchy(double) (HierarchyIntegrator.cpp:
291)                                                                                                  
==21028==    by 0x478646: main (example.cpp:436)
==21028== 
==21028== Invalid read of size 4
==21028==    at 0x7F0D50: operator== (CartSideRobinPhysBdryOp.cpp:490)
==21028==    by 0x7F0D50: IBTK::CartSideRobinPhysBdryOp::setPhysicalBoundaryConditions(SAMRAI::hier::Pa
tch<3>&, double, SAMRAI::hier::IntVector<3> const&) (CartSideRobinPhysBdryOp.cpp:490)                 
==21028==    by 0x7D0298: IBTK::HierarchyGhostCellInterpolation::fillData(double) (HierarchyGhostCellIn
terpolation.cpp:635)                                                                                  
==21028==    by 0x71731B: IBAMR::StaggeredStokesOperator::apply(SAMRAI::solv::SAMRAIVectorReal<3, doubl
e>&, SAMRAI::solv::SAMRAIVectorReal<3, double>&) (StaggeredStokesOperator.cpp:220)                    
==21028==    by 0x717BBD: IBAMR::StaggeredStokesOperator::modifyRhsForBcs(SAMRAI::solv::SAMRAIVectorRea
l<3, double>&) (StaggeredStokesOperator.cpp:381)                                                      
==21028==    by 0x5BDFF4: IBAMR::CIBStaggeredStokesOperator::modifyRhsForBcs(_p_Vec*) (CIBStaggeredStok
esOperator.cpp:316)                                                                                   
==21028==    by 0x56D9BF: IBAMR::CIBSaddlePointSolver::solveSystem(_p_Vec*, _p_Vec*) (CIBSaddlePointSol
ver.cpp:396)                                                                                          
==21028==    by 0x5C22EE: IBAMR::CIBStaggeredStokesSolver::solveSystem(SAMRAI::solv::SAMRAIVectorReal<3
, double>&, SAMRAI::solv::SAMRAIVectorReal<3, double>&) (CIBStaggeredStokesSolver.cpp:282)            
==21028==    by 0x6F4181: IBAMR::INSStaggeredHierarchyIntegrator::integrateHierarchy(double, double, in
t) (INSStaggeredHierarchyIntegrator.cpp:1421)                                                         
==21028==    by 0x5DCD45: IBAMR::IBExplicitHierarchyIntegrator::integrateHierarchy(double, double, int)
 (IBExplicitHierarchyIntegrator.cpp:283)                                                              
==21028==    by 0x955B96: IBTK::HierarchyIntegrator::advanceHierarchy(double) (HierarchyIntegrator.cpp:
291)                                                                                                  
==21028==    by 0x478646: main (example.cpp:436)
==21028==  Address 0x70 is not stack'd, malloc'd or (recently) free'd
==21028== 
[0]PETSC ERROR: ------------------------------------------------------------------------
[0]PETSC ERROR: Caught signal number 11 SEGV: Segmentation Violation, probably memory access out of ran
```
(etc). I believe it is from #880 because, when bisecting, thats when things started going bad.

**Comment by drwells** [comment]
It looks like @amneetb has a fix: see https://github.com/IBAMR/IBAMR/commit/c122fc9ad58cc6eb923b750828239cad151fde5b#r37817845

**Comment by drwells** [comment]
Fixed by #915.
